### ISSUE TITLE
Transactions CSV file inserts tab character before any negative value

### ISSUE THREAD

**Issue by tvolpert** [Automattic/woocommerce-payments#7307]
### Describe the bug
When downloading transactions to a CSV file, any fields with negative values (for example, the entire `Fees` column) have a _tab_ character added - this makes them hard to work with in programs such as Excel and Google sheets (interestingly, Apple's Numbers seems to ignore the tab character)

### To Reproduce
1. Go to `Payments > Transactions` on your site's wp-admin sidebar
3. Click on `Download`
4. Open the CSV file in Excel or Google sheets
5. try to do a `SUM` (or any other math function) with the `Fees` column, and it won't work as expected because of the tabs

### Screenshots
<img width="1370" alt="Screenshot 2023-09-27 at 10 11 07 AM" src="https://github.com/Automattic/woocommerce-payments/assets/34486913/cb14fd8a-ed26-4a6f-8d09-1b5904799a0c">
_showing the tab characters in the raw file by running a regex search in Visual Studio Code_

### Expected behavior
We should be able to do quick calculations within a spreadsheet program for all numerical values 

### Additional context
brought to my attention in 7086428-zen

**Comment by zmaglica** [comment]
Hey there. Can you maybe replace all tabs with spaces using Visual Studio Code and import that into Google Sheets? In the meantime, I will put this issue for prioritization.

**Comment by jessy-p** [comment]
This issue impacts Reporting, so assigning to Helix (based on team responsibilities Pc2DNy-3z-p2) @haszari. Assigning as part of Gamma Triage process PcreKM-yM-p2.

**Comment by haszari** [comment]
Thanks @jessy-p ! As of just now, reporting is now maintained by @Automattic/fusion (see Woo Transact P2 pc2DNy-3fz-p2) – so this and maybe some other issues might need reassignment.

**Comment by anu-rock** [comment]
Moved it to Fusion's backlog.

**Comment by elazzabi** [comment]
I gave this a look.

It looks like we are using `@woocommerce/csv-export`, and after looking at [the implementation](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/csv-export/src/index.ts#L26), I see that this is an expected behavior to prevent CSV injections.

All values starting with '=', '+', '-', or '@', will have a tab character at the start.

**Comment by anu-rock** [comment]
@elazzabi Thanks for looking into this. The [link](http://www.contextis.com/resources/blog/comma-separated-vulnerabilities/) referenced in that code comment seems to be broken. I'm wondering if there are other ways to prevent CSV injections that play well with spreadsheet programs. [Symphony](https://symfony.com/blog/cve-2021-41270-prevent-csv-injection-via-formulas) seems to consider even `\t` as a vulnerable character (based on [OWASP's recommendation](https://owasp.org/www-community/attacks/CSV_Injection)), and uses the `'` prefix instead. Is this something you can quickly test?

**Comment by elazzabi** [comment]
Interesting recommendation by OWASP. Given the package lives in `woocommerce`, it's a little challenging to test it, but doable. I can give it a try.

However, adding a `'` will indeed escape the problematic text, but that will break a few things I think. I imagine many customers will use the built-in Excel functions (ie `SUM`) to summarize fees. This will work with backspace (I think), but when adding ', that will turn the columns to text, which will make it harder to use it.

@tvolpert do you have any idea if `'` will break the current user experience? Or if it's already broken anyway 😭

**Comment by tvolpert** [comment]
@elazzabi I would say it's already broken anyway; the tab character already causes Excel and Google Sheets to recognize the columns as text rather than numbers. The `'` would at least be _visible_, so it would be much easier for the average user to diagnose and correct the issue on their own. With the tab character, you have to have at least a passing familiarity with RegEx to even figure out what that character is to do a search-replace.

**Comment by elazzabi** [comment]
Sounds good! This must be fixed on WC Core, publish a new package, and then update the relevant package here.

I'll start working on this.

**Comment by elazzabi** [comment]
Update:

It seems using the `'` is the way to go here. We even use that in Core: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/export/abstract-wc-csv-exporter.php#L374

I created a PR linked ^, once merged, we'll need to update the package in WooPayments and merge it.

**Comment by anu-rock** [comment]
I've pushed a couple of finishing changes to the ☝️ PR and asked for reviews in p1699960176703109-slack-C025EHY377G.

**Comment by anu-rock** [comment]
@Automattic/harmony We'll need to eventually update the version for `@woocommerce/csv-export` in package.json once the above PR is approved. Do you foresee any challenges in doing so? From what I can see, there hasn't been a new version of this [package](https://www.npmjs.com/package/@woocommerce/csv-export) after 1.7.0 (what we currently use).

**Comment by jbordonado** [comment]
> @Automattic/harmony We'll need to eventually update the version for `@woocommerce/csv-export` in package.json once the above PR is approved. Do you foresee any challenges in doing so? From what I can see, there hasn't been a new version of this [package](https://www.npmjs.com/package/@woocommerce/csv-export) after 1.7.0 (what we currently use).

I don't see us bundling explicitly this package (see our [webpack config](https://github.com/Automattic/woocommerce-payments/blob/c4448072a793b73bab9634522fc2ed081f926911/webpack/shared.js#L137)) via the Webpack Dependency Extraction Plugin.

Assuming that this package ships with WooCommerce (which is the case for almost all of them if I recall correctly), it would just be a matter of updating the version in our package.json and ensuring nothing is broken in our tests. It won't change what's actually being used in production, as we'll still use whatever is shipped directly inside WooCommerce and exposed in the window. More details about what I mean here if that's not clear: paJDYF-8M2-p2#comment-18369

Also, we have a doc to track the dependencies updates: https://github.com/Automattic/woocommerce-payments/blob/develop/docs/dependencies.md#javascript-dev-dependencies

If you update that one, please add a row with the relevant details for the next iterations

**Comment by anu-rock** [comment]
Ah, totally missed that. Thanks for clarifying again, @jbordonado 🙌

**Comment by anu-rock** [comment]
> Also, we have a doc to track the dependencies updates: https://github.com/Automattic/woocommerce-payments/blob/develop/docs/dependencies.md#javascript-dev-dependencies

@jbordonado For some reason, the CSV package is not listed here. Should I add it?

**Comment by jbordonado** [comment]
> > Also, we have a doc to track the dependencies updates: https://github.com/Automattic/woocommerce-payments/blob/develop/docs/dependencies.md#javascript-dev-dependencies
> 
> @jbordonado For some reason, the CSV package is not listed here. Should I add it?

Yes, that's what I was saying in my earlier comment.
That package has seemingly never been updated since it was first introduced, so it's not part of the table yet.

**Comment by anu-rock** [comment]
**Update:**
* The Woo Core PR https://github.com/woocommerce/woocommerce/pull/41163 is merged now 🎉 
* I will create another PR to update the CSV package version in WooPayments.

**Comment by anu-rock** [comment]
WooPayments PR created at https://github.com/Automattic/woocommerce-payments/pull/7729. Some tests are failing because a new version of upstream package has not yet been released. I'll come back for reviews once that happens.

**Comment by Jinksi** [comment]
I'm revisiting the escaping of negative CSV values for #9303 and have proposed removing the escaping for pure numeric values.

I'm proposing an alternative solution in https://github.com/woocommerce/woocommerce/pull/52727 which will **allow negative numeric values to remain unescaped in the CSV output** since a pure numeric value cannot form a valid formula.

@elazzabi @anu-rock Based on your discussion and prior work here, are there considerations we should be aware of before allowing negative numeric values through unescaped?

**Comment by elazzabi** [comment]
Hey @Jinksi 👋 from [OWASP](https://owasp.org/www-community/attacks/CSV_Injection) and some [GPT searching](https://chatgpt.com/share/673306b8-1c70-800b-8c95-87cc47dc7044) it looks like some applications can still interpret it as a formula and execute it. Things like `-A1` or `-1+2`, etc...

**Comment by haszari** [comment]
> from [OWASP](https://owasp.org/www-community/attacks/CSV_Injection) and some [GPT searching](https://chatgpt.com/share/673306b8-1c70-800b-8c95-87cc47dc7044) it looks like some applications can still interpret it as a formula and execute it. Things like -A1 or -1+2, etc...

@elazzabi  so if we sanitise to ensure that these are pure numeric values, we should be ok right? The examples you share are not simple numbers, I'm hopeful that we can sanitise as per @Jinksi 's idea.

**Comment by Jinksi** [comment]
> Things like `-A1` or `-1+2`, etc...

@haszari @elazzabi Exactly! These values would have to be strings in JS, so would remain escaped in the proposal.

Example of values starting with `-` that are a valid `number` type in JS that would remain unescaped in the proposal:


```js
[
-42,
-3.14159,
-0.00456,
-Infinity,
-9007199254740991 // Number.MIN_SAFE_INTEGER
]
```

Since the unescaped `number.toString()` result would be the entire of a cell's value, there is no chance of a formula being injected into this cell.

**Comment by elazzabi** [comment]
> @haszari @elazzabi Exactly! These values would have to be strings in JS, so would remain escaped in the proposal.

Aha! That makes sense. If we make sure it's a number, then I think we should be okay 🙌 

Thanks for working on this!
